pkg/tfshim/sdk-v2/provider2.go

```diff
@@ -326,11 +327,7 @@ func (p *planResourceChangeImpl) upgradeState(
 
 	ty := res.CoreConfigSchema().ImpliedType()
 
-	jsonIsh, err := schema.StateValueToJSONMap(state.stateValue, ty)
-	if err != nil {
-		return nil, err
-	}
-	jsonBytes, err := json.Marshal(jsonIsh)
+	jsonBytes, err := ctyjson.Marshal(state.stateValue, ty)
```

Comment:
Can you explain the change here - seems like this is cutting out a bunch of code, which looks good but what was the issue with the previous code?

Reply:
Ah, saw your commit message about it, thanks!